pkg/kubelet/cm/devicemanager/manager.go

```diff
@@ -100,8 +100,9 @@ type ManagerImpl struct {
 }
 
 type endpointInfo struct {
-	e    endpoint
-	opts *pluginapi.DevicePluginOptions
+	e                 endpoint
+	opts              *pluginapi.DevicePluginOptions
+	registrationCount int
```

Comment:
Hmm.. can't you just poll m.healthyDevices[resource] the same way Swati does in her patch?

Reply:
Possibly. The intent here is to capture the "Registration happened". Registration in turn updates `healthyDevices`.
I think the event which best represents that the wait should be over because device plugins are online and connected again is indeed the processing of `ListAndWatch`, which is done as last step of the registration.

I'm surely open about improving this code and perhaps drop any extra variables I added.